pkg/pulumiyaml/run.go

```diff
@@ -1800,8 +1846,17 @@ func (e *programEvaluator) evaluateBuiltinInvoke(t *ast.InvokeExpr) (interface{}
 	var opts []pulumi.InvokeOption
 
 	if t.CallOpts.Version != nil {
-		opts = append(opts, pulumi.Version(t.CallOpts.Version.Value))
+		if versionExpr, ok := e.evaluateExpr(t.CallOpts.Version); ok {
+			if version, ok := versionExpr.(string); ok {
+				opts = append(opts, pulumi.Version(version))
+			} else {
+				e.error(t.CallOpts.Version, "version must be a string value")
+			}
```

Comment:
This should be an internal error. It version resolves to something not a string, then the type checker should have caught it.

Reply:
Yeah added a type-check for `Version` such that it must be assignable to string, so removed this one ✅